activerecord/lib/active_record/database_configurations/hash_config.rb

```diff
@@ -66,6 +66,18 @@ def pool
         (configuration_hash[:pool] || 5).to_i
       end
 
+      def min_threads
+        (configuration_hash[:min_threads] || 0).to_i
+      end
+
+      def max_threads
+        (configuration_hash[:max_threads] || pool).to_i
```

Comment:
I suppose we have until Rails 7 release to figure good defaults, so until then this is ok.

But I think we'll have to find something better. As @fxn and other people pointed, right sizing the pool will now be a bit more involved.

Until now mosts apps have `pool: <%= ENV['RAILS_MAX_THREADS'] %>`, which is fine and simple. 

Now you'd need `pool: <%= ENV['RAILS_MAX_THREADS'].to_i + global_executor_concurrency %>`. 

But maybe all we'll need would be to document this properly.

Reply:
Since #31221, I don't think people need to configure their pool size so carefully... it's less "this is the number of connections each process should and will consume", and more like a final breakpoint to prevent a single process consuming every connection up to your DB's configured server-wide limit.

The above are definitely still reasonable minimum values.. but it's a lot safer to set a much higher limit by default than it once was. I actually think we could and should consider defaulting to an unlimited pool.

Reply:
What I have mostly in mind is related to documentation. Thinks like:

1. What does the upgrade guide recommends.
2. What does the docs about configuring Rails suggest.
3. What does new app generators generate.

Reply:
>  I actually think we could and should consider defaulting to an unlimited pool.

Oh! That does indeed sound like an interesting idea.

Reply:
Like the idea too.

Normally you do not configure the pool for contention, but after the concurrency settings of your web server, Sidekiq, etc. Makes sense, then, that those settings are the ones that define the real number of connections established as a side-effect of the runtime.